qemu/tests/convert_to_virtual_disk.py

```diff
@@ -0,0 +1,47 @@
+import logging
+
+from avocado import fail_on
+from avocado.utils import process
+
+from virttest import data_dir
+from virttest.qemu_storage import QemuImg
+
+
+def run(test, params, env):
+    """
+    Convert a image to a virtual block device.
+    1. Create input and output images.
+    2. Setup loop device with the output image.
+    3. Convert the input image to the loop device.
+    :param test: Qemu test object.
+    :param params: Dictionary with the test parameters.
+    :param env: Dictionary with test environment.
+    """
+    input_img = params["images"].split()[0]
+    output_img = params["images"].split()[-1]
+    root_dir = data_dir.get_data_dir()
+    source = QemuImg(params.object_params(input_img), root_dir, input_img)
+    output = QemuImg(params.object_params(output_img), root_dir, output_img)
+    output_filename = output.image_filename
+
+    logging.info("Create the test image files.")
+    source.create(source.params)
+    output.create(output.params)
+
+    logging.info("Setup target loop device via 'losetup'.")
+    try:
+        process.run("losetup -f %s" % output_filename)
+    except process.CmdError as err:
+        test.error("Command line failed: %s" % err)
+
+    target = process.system_output("losetup -l | tail -n1 | awk '{print $1}'",
+                                   shell=True).decode()
```

Comment:
It cannot be guaranteed the lastest loop device is the one created, better to search by the back-file.

Reply:
Please wrap the functionality to probe loop-device by back-file into a function.